src/stream/src/from_proto/chain.rs

```diff
@@ -172,6 +174,10 @@ impl ExecutorBuilder for ChainExecutorBuilder {
             }
             ChainType::ChainUnspecified => unreachable!(),
         };
-        Ok(executor)
+        if let Ok(rate_limit) = node.get_rate_limit() {
+            Ok(FlowControlExecutor::new(executor, *rate_limit).boxed())
```

Comment:
To clarify, eventually the executors will be like: (`->` means the direction of data flow)

```
upstream MaterializeExecutor -> BackfillExecutor -> FlowControlExecutor -> ...
```

Is it correct?

If so, the `FlowControlExecutor` will also limit the event stream from upstream. Looks good to me but need to be noted.

Reply:
Okay added it as a docstring to `FlowControlExecutor`.

Reply:
Something to note: `FlowControlExecutor` MUST come after `BackfillExecutor`, otherwise it will only rate limit update messages.